components/brave_wallet/browser/keyring_service.cc

```diff
@@ -521,11 +521,12 @@ void KeyringService::RequestUnlock() {
 
 HDKeyring* KeyringService::ResumeKeyring(const std::string& keyring_id,
                                          const std::string& password) {
-  if (!CreateEncryptorForKeyring(password, keyring_id)) {
+  if (!encryptor_ && !CreateEncryptorForKeyring(password, keyring_id)) {
```

Comment:
Encryptor and its salt is designed to be used per keyring, so different keyring can't share the same encryptor
We can share same password but not encryptor, use map to store encryptor per keyring keyed by keyring id

Reply:
done